File: src/242ValidAnagram/242ValidAnagram.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
  bool isAnagram1(string s, string t) {
    sort(s.begin(), s.end());
    sort(t.begin(), t.end());
    return s == t;
  }
  bool isAnagram(string s, string t) {
    unordered_map<char, int> m;
    for (auto &c : s) {
      m[c]++;
    }
    for (auto &c : t) {
      m[c]--;
    }
    for (auto &p : m) {
      if (p.second != 0) {
        return false;
      }
    }
    return true;
  }
};

int main() {
  Solution s;
  cout << s.isAnagram("anagram", "nagaram") << endl;
  cout << s.isAnagram("anagramb", "nagaram") << endl;
  cout << s.isAnagram("rat", "car") << endl;
  return 0;
}

File: src/242ValidAnagram/242ValidAnagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {return false;}
        let mut count = vec![0;128];
        let si = s.into_bytes();
        let ti = t.into_bytes();
        for c in si{
            count[c as usize]+=1;
        }
        for c in ti{
            count[c as usize]-=1;
            if count[c as usize]<0 {
                return false;
            }
        }
        true
    }
}
